internal/state/dataplane/configuration_test.go

```diff
@@ -1393,9 +1393,12 @@ func TestBuildUpstreams(t *testing.T) {
 
 	g := NewGomegaWithT(t)
 
-	upstreams := buildUpstreamsMap(context.TODO(), listeners, fakeResolver)
+	upstreams := buildUpstreams(context.TODO(), listeners, fakeResolver)
+	sort.Slice(upstreams, func(i, j int) bool {
+		return upstreams[i].Name < upstreams[j].Name
+	})
 
-	g.Expect(helpers.Diff(upstreams, expUpstreams)).To(BeEmpty())
+	g.Expect(helpers.Diff(expUpstreams, upstreams)).To(BeEmpty())
```

Comment:
Could this work here? https://onsi.github.io/gomega/#containelementselement-interface 
( without making parsing failure cases output harder )

Reply:
Yep, I totally forgot about that matcher. Ty!